File: src/main/java/de/check24/ui/pages/base/BasePage.java
package de.check24.ui.pages.base;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import static de.check24.ui.driver.Driver.click;
import static de.check24.ui.driver.Driver.navigateTo;

public class BasePage {
    protected static final Logger log = LoggerFactory.getLogger(BasePage.class);

    protected final String BASE_URL = "https://www.check24.de/";

    private final String COOKIE_CONSENT_BUTTON = "//div[contains(@class, 'c24-cookie-consent-notice-buttons')]//a[@class='c24-cookie-consent-button']";

    public void navigateToHomePage() {
        navigateTo(BASE_URL);
    }

    public void clickCookieConsentButton() {
        click(COOKIE_CONSENT_BUTTON);
    }
}

File: js/test/pageobjects/base.page.js
import { browser } from '@wdio/globals'

/**
* main page object containing all methods, selectors and functionality
* that is shared across all page objects
*/
export default class BasePage {
    /**
    * Opens a sub page of the page
    * @param path path of the sub page (e.g. /path/to/page.html)
    */
    open (path) {
        return browser.url("https://www.check24.de/");
    }

    closeCookies () {
        const cookieConsentButton = $("//div[contains(@class, 'c24-cookie-consent-notice-buttons')]//a[@class='c24-cookie-consent-button']");  
        cookieConsentButton.click()
    }

    fill(locator, value) {
       $(locator).setValue(value);
    }

    async click(locator) {
        await $(locator).click();
    }

    async waitAndClick(locator) {
        const element = await $(locator);
        await element.waitForClickable({timeout: 15000});
        await element.click();
    }

    async getElementText(locator) {
        return await $(locator).getText()
    }

    async getElementList(locator) {
        await $(locator).waitForExist({timeout: 10000});
        return await $$(locator);
    }

    async getQuantityOfElements(locator) {
        const elements = await $$(locator);
        return elements.length;
    }
}
